test/suite/test_cursor13.py

```diff
@@ -42,12 +42,35 @@ class test_cursor13_base(wttest.WiredTigerTestCase):
     stat_cursor_reopen = 0
 
     # Returns a list: [cursor_cached, cursor_reopened]
+    #
+    # We want the statistics for operations triggered from our program. The challenge is that
+    # eviction threads may cache history store cursors in the background. We address this by
+    # subtracting out operations from the history store file. This is tricky because we can't
+    # atomically check the connections stats and the history store stats.  So we look at the
+    # history store stats before and after the connection stats and only accept a result where
+    # the history store stats haven't changed.
     def caching_stats(self):
-        stat_cursor = self.session.open_cursor('statistics:', None, None)
-        cache = stat_cursor[stat.conn.cursor_cache][2]
-        reopen = stat_cursor[stat.conn.cursor_reopen][2]
-        stat_cursor.close()
-        return [cache, reopen]
+        hs_stats_uri = 'statistics:file:WiredTigerHS.wt'
+        while 1:
```

Comment:
Although this has been merged, is there any chance this could become an infinite loop? Would we want to put a maximum on this? Also should this be `while True:` instead of 1?

Reply:
Good point.  I think it is pretty unlikely this would loop forever, but it's better to code it defensively now, rather than have some engineer scratching their head over a hung test case some time in the future. I'll open a new ticket and make the change.  (And then give you another PR to review :-)